src/test/scala/chiselTests/TransitNameSpec.scala

```diff
@@ -0,0 +1,54 @@
+package chiselTests
+
+import org.scalatest.{FlatSpec, Matchers}
+
+import chisel3._
+import chisel3.experimental.RawModule
+import chisel3.util.TransitName
+
+import firrtl.FirrtlExecutionSuccess
+
+class TransitNameSpec extends FlatSpec with Matchers {
+
+  class MyModule extends RawModule {
+    val io = IO(new Bundle{})
+    override val desiredName: String = "MyModule"
+  }
+
+  /** A top-level module that instantiates three copies of MyModule */
+  class Top extends RawModule {
+
+    /* Assign the IO of a new MyModule instance to value "foo". The instance will be named "MyModule". */
+    val foo = Module(new MyModule).io
+
+    /* Assign the IO of a new MyModule instance to value "bar". The instance will be named "bar". */
+    val bar = {
+      val x = Module(new MyModule)
+      TransitName(x.io, x)
```

Comment:
Not to nitpick too much on this:  would it be clearer to write it this way for new users so they don't need to guess at what TransitName returns?
```scala
    val bar = {
      val x = Module(new MyModule)
      TransitName(x.io, x)
      x.io
    }
```

Reply:
All good. Probably. `TransitName` [has been brought up on SO as confusing before](https://stackoverflow.com/questions/49863120/what-is-the-purpose-of-transitname-in-chisel3). Nonetheless, it's one usage (in the `Queue` factory) does it like this. I'm more in favor of the alternative approach so that users are less likely to be confused if they read other people's code.